src/librustc_mir/transform/const_prop.rs

```diff
@@ -380,22 +379,23 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
             source_scopes: body.source_scopes.clone(),
             //FIXME(wesleywiser) we can't steal this because `Visitor::super_visit_body()` needs it
             local_decls: body.local_decls.clone(),
-            ret: ret.map(Into::into),
             source_info: None,
         }
     }
 
     fn get_const(&self, local: Local) -> Option<OpTy<'tcx>> {
+        let op = self.ecx.access_local(self.ecx.frame(), local, None).ok();
+
         if local == RETURN_PLACE {
```

Comment:
I don't understand why const-prop needs a special case here. So far there shouldn't even have been any reads from the return place in the generated MIR?

Reply:
Looking at the code, seems like the goal is to normalize to `Operand::Immediate` where possible. That makes sense, I guess, if the later code prefers that over an `Indirect`.

What I do not understand is what that is done only for the return place. My guess is that previously the return place needed a special path anyway, and then for some reason someone added this normalization only there. I think now we should just always do it.

Reply:
Huh, that seems to work just fine. It makes const prop handle even more cases (causing an incremental test to fail, but that was easy to fix).